engine/core/src/main/java/org/datacleaner/metamodel/datahub/DataHubDataSet.java

```diff
@@ -122,20 +125,22 @@ public boolean next() {
 
         _nextPageFirstRow = _nextPageFirstRow + _nextPageMaxRows;
 
-        String uri = _uri + createParams(firstRow, maxRows);
+        final String uri = _uri + createParams(firstRow, maxRows);
 
-        HttpGet request = new HttpGet(uri);
+        final HttpGet request = new HttpGet(uri);
         request.addHeader(ACCEPT, JSON_CONTENT_TYPE);
-        HttpResponse response = executeRequest(request);
-
-        return getResultSet(response.getEntity());
+        final HttpResponse response = executeRequest(request);
+        final List<Object[]> resultSet = getResultSet(response.getEntity());
+        final int resultSetSize = resultSet.size();
+        _endReached = (resultSetSize < maxRows);
```

Comment:
I don't think so. Look at line 124. The maxRows is taken from the query, or becomes _nextPageMaxRows when not set in the query. By using maxRows, my intention is to also make it work when the query is setting a maxRows by itself.